src/environment.rs

```diff
@@ -57,7 +57,7 @@ impl OsStrExtLmdb for OsStr {
 /// An environment supports multiple databases, all residing in the same shared-memory map.
 pub struct Environment {
     env: *mut ffi::MDB_env,
-    dbi_open_mutex: Mutex<()>,
+    env_mutex: RwLock<()>,
```

Comment:
Would it make sense to enclose the `MDB_env` pointer *inside* the `RwLock` to somehow suggest it's supposed to be accessed with the lock acquired? Not sure it makes much sense since you can still copy the pointer out of the lock but at least it will raise the bar accessing the pointer without a lock.

Reply:
Not really... some times it's needed without lock too for unsafe impls. This setting minimizes redundancy. Any other setting will incur extra, unnecessary cost.